test/test_torch.py

```diff
@@ -3772,6 +3772,39 @@ def get_probs(length, is_contiguous):
                 a_t, p_t = torch._multinomial_alias_setup(probs)
                 torch._multinomial_alias_draw(p_t.view(2, 2), a_t.view(2, 2))
 
+        MAX_SAMPLES = 100000
+        for probs in [get_probs(4, True),
+                      cast(torch.tensor([0.8, 0.2])),
+                      cast(torch.tensor([0.7, 0.2, 0.1]))]:
+            # Check how different the alias distribution and the original distribution are
+            n_samples = 0
+            alias_dist = torch.zeros_like(probs)
+            alias_table, prob_table = torch._multinomial_alias_setup(probs)
+            while n_samples < MAX_SAMPLES:
+                n_samples += 10000
+                alias_samples = torch._multinomial_alias_draw(prob_table, alias_table, 10000)
```

Comment:
This needs to be `n_samples` too instead of 10000. It is failing on CUDA, though for a more fundamental reason, since it seems like it is just drawing from a uniform distribution instead. I think it might be better to simplify and use MAX_SAMPLES by default instead of running this in a loop.

Since you haven't modified the CUDA sampler, maybe this was already an issue with the CUDA implementation? In that case, feel free to feel free to skip this test on CUDA, but we should open another issue to track this and resolve it separately.

```
18:51:40 ======================================================================
18:51:40 FAIL: test_multinomial_alias (__main__.TestCuda)
18:51:40 ----------------------------------------------------------------------
18:51:40 Traceback (most recent call last):
18:51:40   File "/var/lib/jenkins/workspace/test/common_utils.py", line 358, in wrapper
18:51:40     method(*args, **kwargs)
18:51:40   File "test_cuda.py", line 2353, in test_multinomial_alias
18:51:40     _TestTorchMixin._test_multinomial_alias(self, lambda t: t.cuda())
18:51:40   File "/var/lib/jenkins/workspace/test/test_torch.py", line 3836, in _test_multinomial_alias
18:51:40     "Actual: {}\nExpected: {}".format(alias_dist, probs))
18:51:40 AssertionError: False is not true : Actual: tensor([24953., 24901., 25088., 25058.], device='cuda:0')
18:51:40 Expected: tensor([0.2007, 0.4799, 0.2581, 0.0613], device='cuda:0')
```

Reply:
Since samples are IID, I thought that accumulating the counts while sampling a smaller number of samples would reduce time. This is the same logic used in the KL divergence tests.

But yes, the problem is pretty fundamental w.r.t. the CUDA sampler. I’ll add a skipIf and create an issue to track this issue.